test/camera/camera_test.py

```diff
@@ -0,0 +1,391 @@
+
+        calibrate_pan_test(camera, 181, 180)
+
+
+    def test_calibrate_pan_realtive_decrease(self):
+        camera = Camera(camera_type=RELATIVE)
+
+        calibrate_pan_test(camera, -10, -10)
+
+
+    def test_calibrate_pan_realtive_decrease_at_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+
+        calibrate_pan_test(camera, -180, -180)
+
+
+    def test_calibrate_pan_realtive_decrease_beyond_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+
+        calibrate_pan_test(camera, -181, -180)
+
+
+    def test_calibrate_tilt_increase_absolute(self):
+        camera = Camera()
+
+        calibrate_tilt_test(camera, 10, 10)
+
+
+    def test_calibrate_tilt_absolute_increase_at_bound(self):
+        camera = Camera()
+
+        calibrate_tilt_test(camera, 150, 150)
+
+
+    def test_calibrate_tilt_absolute_increase_beyond_bound(self):
+        camera = Camera()
+
+        calibrate_tilt_test(camera, 151, 150)
+
+
+    def test_calibrate_tilt_decrease_absolute(self):
+        camera = Camera()
+
+        calibrate_tilt_test(camera, -10, -10)
+
+
+    def test_calibrate_tilt_absolute_decrease_at_bound(self):
+        camera = Camera()
+
+        calibrate_tilt_test(camera, -150, -150)
+
+
+    def test_calibrate_tilt_absolute_decrease_beyond_bound(self):
+        camera = Camera()
+
+        calibrate_tilt_test(camera, -151, -150)
+
+
+    def test_calibrate_tilt_relative_increase(self):
+        camera = Camera(camera_type=RELATIVE)
+
+        calibrate_tilt_test(camera, 10, 10)
+
+
+    def test_calibrate_tilt_relative_increase_at_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+
+        calibrate_tilt_test(camera, 150, 150)
+
+
+    def test_calibrate_tilt_relative_increase_beyond_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+
+        calibrate_tilt_test(camera, 151, 150)
+
+
+    def test_calibrate_tilt_relative_decrease(self):
+        camera = Camera(camera_type=RELATIVE)
+
+        calibrate_tilt_test(camera, -10, -10)
+
+
+    def test_calibrate_tilt_relative_decrease_at_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+
+        calibrate_tilt_test(camera, -150, -150)
+
+
+    def test_calibrate_tilt_relative_decrease_beyond_bound(self):
+        camera = Camera(camera_type=RELATIVE)
+
+        calibrate_tilt_test(camera, -151, -150)
+
+
+    def test_reset_camera_absolute(self):
+        camera = Camera()
+        camera.pan(150)
+        camera.tilt(90)
+
+        camera.reset_camera()
+
+        self.assertEqual(camera._current_pan_angle, 90)
+        self.assertEqual(camera._current_tilt_angle, 0)
+
+
+    def test_reset_camera_realtive(self):
+        camera = Camera(camera_type=RELATIVE)
+        camera.pan(60)
+        camera.tilt(90)
+
+        camera.reset_camera()
+
+        self.assertEqual(camera._current_pan_angle, 90)
+        self.assertEqual(camera._current_tilt_angle, 0)
+
+
+def calibrate_pan_test(camera, calibration_value, expected_value):
```

Comment:
Why aren't these a part of the class?

Reply:
I was not exactly sure the correct way to call in the object and am unable to test the specific structure in unittest without the picar. I got it to work in a sudo test file outside of the class, let me know if there are any errors or what happens if you get time to run it. thanks.